Serpent/Serpent/Classes/Extensions/URLSession+Codable.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  URLSession+Codable.swift
+//  Serpent
+//
+//  Created by Andrei Hogea on 16/03/2018.
+//  Copyright © 2018 Nodes. All rights reserved.
+//
+
+import Foundation
+
+// Decoded Result
+public enum DResult<Value> {
+    case success(Value)
+    case failure(Error)
+}
+
+public extension URLSession {
+    
+    /**
+     Adds a handler that attempts to parse the result of the request into a **Decodable**
+     
+     - parameter requestCompletion: The URLSession.dataTask completion
+     
+     - returns: The Decoded Result (DResult)
+     */
+    public func decode<Value: Swift.Decodable>(requestCompletion: (Data?, Error?)) -> DResult<Value> {
+        let (data, error) = requestCompletion
```

Comment:
I'm just nitpicking here, but wouldn't it look better if we did a `switch` on this, instead of doing multiple `if let`s? Something like this:

```
let closure: (Data?, Error?) = (nil, nil)

switch closure {
case (.some(let data), .some(let error)):
    print("got data \(data.hashValue) and error \(error.localizedDescription)")
    
case (.some(let data), .none):
    print("got data \(data.hashValue)")
    
case (.none, .some(let error)):
    print("got error \(error.localizedDescription)")
    
case (.none, .none):
    print("nil")
}
```